lib/build.py

```diff
@@ -254,6 +254,18 @@ def __init__(self, tree):
             ch.FATAL("%3d %s: repeated option --%s"
                      % (self.lineno, self.str_name, k))
          self.options[k] = v
+
+      for st in ch.tree_children(tree, "option_keypair"):
+         k = ch.tree_terminal(st, "OPTION_KEY")
+         s = ch.tree_terminal(st, "OPTION_VAR")
+         v = ch.tree_terminal(st, "OPTION_VALUE")
+         # assuming all key pair options allow multiple options
+         ch.INFO("setting %s to %s" % (s, v))
+         try:
```

Comment:
I'm not personally a fan of the Python ask-forgiveness approach, because it seems hard to reason about whether the `KeyError` came from the dictionary lookup I was intending. So, in this case I'd do `if (k in options):`.

However: I think here you can do a one-liner:

```
self.options.setdefault(k, {})[s] = v
```